content/blog/zardshard/2023-05-05_[gsoc_2023]_plans_for_improving_icon-o-matic.md

```diff
@@ -0,0 +1,47 @@
++++
+type = "blog"
+title = "[GSoC 2023] Plans for improving Icon-O-Matic"
+author = "Zardshard"
+date = "2023-05-05 14:07:09-04:00"
+tags = ["haiku", "software", "icon-o-matic", "gsoc", "gsoc2023"]
++++
+
+Icon-O-Matic is the editor for HVIF files, the format that Haiku's icons are in. It's a relatively simple but impressive aplication. It does have room for improvement, however.
+
+There are three things I am planning to do during GSoC: fix bugs, refactor the code, and improve the UI. There is a list of known bugs that Icon-O-Matic has over on [Trac](https://dev.haiku-os.org/query?component=%5EApplications%2FIcon-O-Matic&status=assigned&status=in-progress&status=new&status=reopened&order=priority&col=id&col=summary&col=status&col=type&col=priority&col=milestone&col=component). You can help, too, by reporting bugs in Icon-O-Matic over there. Less obviously, you can also vote for tickets that you would like to be resolved. To do this, click on the ticket that interests you, then click one of the buttons highlighted by the arrow: ![The buttons are in the upper-right hand corner among the navigation links](/files/blog/zardshard/trac_upvoting.png)
+
+## My current plans
+
+Here are the plans that I currently have. As with all plans, they are subject to change.
+
+### Add reference images
+
+A feature that would be very nice for Icon-O-Matic to have is reference images. People have asked for this in [bug #2748](https://dev.haiku-os.org/ticket/2748) and [bug #8758](https://dev.haiku-os.org/ticket/8758).
+
+### Add previews to shapes
```

Comment:
Do you mean "Paths"?
I don't quite know what your target is here.
Either you create a new Path, and start clicking into the canvas to set points etc. While you're creating the pth, you see its line. Only when you deselect the path object, it becomes invisible.
Or you create a shape+path+style. Then you select the Path and start setting points etc. While you do that, the shape evolves in the new style. As this can be awkward sometimes, One can add a Stroke as Transformer to only have an outline.

Reply:
Previews of the shapes directly in the shapes list would be interesting, I think Wonderbrush (which is essentially a larger and more complete version of Icon O Matic) does this for layers.

Previews of colors and gradients in the styles list would also be nice to have.

Reply:
No, I mean the shapes listed in the shapes menu. If, for example, you were trying to find out which shape makes the nozzle of the spray can of the debugger icon, you can use the previews to find a black circle. Then, you can click on it to see if it is the black circle you wanted. From there, the path and the style are highlighted, and you can adjust either as you see fit.

Reply:
Just saw pulkomandy's comment. Yes, I think previews for the styles and maybe even for the paths would be nice.

Reply:
I misunderstood. Yes. Previews in the lists would be nice!
I think all comments have been addressed. Let's merge.